pages/cnpj-generator.vue

```diff
@@ -0,0 +1,159 @@
+<template>
+  <v-layout column justify-center align-center>
+    <v-snackbar v-model="snackbar" class="cnpj-generator__snack-bar">
+      Cnpj copied to clipboard!
+    </v-snackbar>
+    <h1 class="font-weight-thin mb-5">Cnpj Generator</h1>
+    <div class="my-2">
+      <v-btn class="cnpj-generator__button" @click="generateCnpj()"
+        >Generate</v-btn
+      >
+    </div>
+    <v-flex class="cnpj-generator__result-wrapper">
+      <p class="cnpj-generator__result">
+        Result:
+        <span v-if="result.cnpj"
+          >{{ result.cnpj }}
+          <v-icon
+            class="cnpj-generator__copy-button"
+            @click="saveToClipBoard(result.cnpj)"
+            >mdi-content-copy</v-icon
+          >
+        </span>
+      </p>
+      <p class="cnpj-generator__result">
+        Result with mask:
+        <span v-if="result.maskCnpj"
+          >{{ result.maskCnpj }}
+          <v-icon
+            class="cnpj-generator__copy-button"
+            @click="saveToClipBoard(result.maskCnpj)"
+            >mdi-content-copy</v-icon
+          >
+        </span>
+      </p>
+    </v-flex>
+  </v-layout>
+</template>
+
+<script>
+import Vue from 'vue'
+
+export default Vue.extend({
+  name: 'CnpjGenerator',
+  data: () => ({
+    result: { cnpj: '', maskCnpj: '' },
+    snackbar: false
+  }),
+  computed: {},
+
+  methods: {
+    saveToClipBoard(text) {
+      navigator.clipboard.writeText(text)
+      this.snackbar = true
+    },
+
+    random(n) {
+      return Math.round(Math.random() * n)
+    },
+
+    mod(dividend, divider) {
+      return Math.round(dividend - Math.floor(dividend / divider) * divider)
+    },
+
+    generateCnpj() {
+      const n = 9
+      const [n1, n2, n3, n4, n5, n6, n7, n8, n9, n10, n11, n12] = new Array(8)
+        .fill(0)
+        .map(() => this.random(n))
+        .concat([0, 0, 0, 1])
+      let d1 =
+        n12 * 2 +
+        n11 * 3 +
+        n10 * 4 +
+        n9 * 5 +
+        n8 * 6 +
+        n7 * 7 +
+        n6 * 8 +
+        n5 * 9 +
+        n4 * 2 +
+        n3 * 3 +
+        n2 * 4 +
+        n1 * 5
+      d1 = 11 - this.mod(d1, 11)
+      if (d1 >= 10) d1 = 0
+      let d2 =
+        d1 * 2 +
+        n12 * 3 +
+        n11 * 4 +
+        n10 * 5 +
+        n9 * 6 +
+        n8 * 7 +
+        n7 * 8 +
+        n6 * 9 +
+        n5 * 2 +
+        n4 * 3 +
+        n3 * 4 +
+        n2 * 5 +
+        n1 * 6
+      d2 = 11 - this.mod(d2, 11)
+      if (d2 >= 10) d2 = 0
+      const cnpj =
+        '' +
+        n1 +
+        n2 +
+        '.' +
+        n3 +
+        n4 +
+        n5 +
+        '.' +
+        n6 +
+        n7 +
+        n8 +
+        '/' +
+        n9 +
+        n10 +
+        n11 +
+        n12 +
+        '-' +
+        d1 +
+        d2
+      this.result.cnpj = cnpj.replace(/[^\w\s]/gi, '')
+      this.result.maskCnpj = cnpj
+    }
```

Comment:
This function can still be improved. It turns out that the implementation is incorrect when you have to compare the mod size. I used some docs as I referenced in the code below:

```suggestion
    /**
     * @see https://www.geradorcnpj.com/algoritmo_do_cnpj.htm
     */
    generateCnpj() {
      const weightsFirstDigit = [5, 4, 3, 2, 9, 8, 7, 6, 5, 4, 3, 2]
      const weightsSecondDigit = [6, 5, 4, 3, 2, 9, 8, 7, 6, 5, 4, 3, 2]
      const digits = new Array(8)
        .fill(9)
        .map(random)
        .concat([0, 0, 0, 1])

      const sumUp = (digits, weights) =>
        digits
          .map((d, index) => d * weights[index])
          .reduce((acc, curr) => (acc += curr), 0)

      const firstMod = mod(sumUp(digits, weightsFirstDigit), 11)
      const d1 = firstMod < 2 ? 0 : 11 - firstMod
      const secondMod = mod(sumUp(digits.concat([d1]), weightsSecondDigit), 11)
      const d2 = secondMod < 2 ? 0 : 11 - secondMod

      const cnpj = digits.concat([d1, d2])

      this.result.cnpj = cnpj.join('')
      this.result.maskCnpj = cnpj
        .join('')
        .replace(/(\d{2})(\d{3})(\d{3})(\d{4})/g, '$1.$2.$3/$4-')
```